### PR TITLE
fix(midi): preserve raw parser state across chunks

### PR DESCRIPTION
**Pull request by danielraffel**
## Summary
- preserve raw MIDI short-message state and running status across transport chunks
- keep realtime bytes transparent while pending short messages and SysEx are in flight
- tighten RPN/NRPN selection handling, including null-selector and mixed-selector edge cases

## Tests
- cmake -S . -B build -DCMAKE_BUILD_TYPE=Debug -DPULP_ENABLE_GPU=OFF
- cmake --build build --target pulp-test-raw-midi-parser pulp-test-midi-expansion --parallel
- ./build/test/pulp-test-raw-midi-parser --reporter compact
- ./build/test/pulp-test-midi-expansion --reporter compact
- git diff --check origin/main...HEAD
- python3 tools/import-validation/check-source-contracts.py --strict
- python3 tools/import-validation/test_source_contracts.py
- python3 tools/scripts/skill_sync_check.py --mode=report --base origin/main --config tools/scripts/versioning.json
- python3 tools/scripts/version_bump_check.py --mode=report

### CODE REVIEW

File: core/midi/include/pulp/midi/rpn_parser.hpp [hunk 1]
```diff
@@ -136,6 +130,38 @@ class RpnParser {
...
+                                   bool is_rpn,
+                                   bool is_msb,
+                                   uint8_t value) {
+        if (state.is_rpn != is_rpn) {
+            state.param_msb = 0;
+            state.param_lsb = 0;
+            state.param_msb_set = false;
+            state.param_lsb_set = false;
+            state.param_set = false;
+            state.value_msb = 0;
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply core/midi/include/pulp/midi/rpn_parser.hpp]: Fixed in fd0191bd8. RPN/NRPN selector mode switches now clear only selector state and preserve the current Data Entry MSB. Added a regression that covers RPN→NRPN and NRPN→RPN mode switches. Validated with cmake --build build --target pulp-test-midi-expansion --parallel and ./build/test/pulp-test-midi-expansion --reporter compact.
